### PR TITLE
Stabilize workspace with parent test

### PR DESCRIPTION
**Pull request by musienko-maxim**
### What does this PR do?
Stabilize the **WorkspaceWithParent** and implement a new solution for disabling the TrustDialog using a new method. The main idea - is to switch it off in background mode with a special JS script using JavascriptExecutor and patching the local database in the browser session. It is similar to setting up its `"security.workspace.trust.enabled":false` with user preferences. Handling the dialog by UI is hard because it is not easy to predict where it can appear.

### Screenshot/screencast of this PR

```
    ✔ Check expected environment variables
          ▼ Dashboard.openDashboard
            ‣ DriverHelper.navigateToUrl
            ‣ DriverHelper.getDriver
          ▼ Dashboard.waitPage
            ‣ DriverHelper.waitVisibility - By(xpath, //div[@id="page-sidebar"]//a[contains(text(), "Workspaces (")])
            ‣ DriverHelper.waitVisibility - polling timed out attempt #1, retrying with 1000ms timeout
            ‣ DriverHelper.waitVisibility - polling timed out attempt #2, retrying with 1000ms timeout
            ‣ DriverHelper.waitVisibility - element is located and is visible.
            ‣ DriverHelper.waitVisibility - By(xpath, //div[@id="page-sidebar"]//a[text()="Create Workspace"])
            ‣ DriverHelper.waitVisibility - element is located and is visible.
            ‣ BrowserTabsUtil.closeAllTabsExceptCurrent
          ▼ BrowserTabsUtil.getAllWindowHandles
            ‣ DriverHelper.getDriver
          ▼ BrowserTabsUtil.getCurrentWindowHandle
            ‣ DriverHelper.getDriver
          ▼ BrowserTabsUtil.switchToWindow
            ‣ DriverHelper.getDriver
          ▼ TestWorkspaceUtil.stopAndDeleteWorkspaceByName
          ▼ TestWorkspaceUtil.stopWorkspaceByName - sample-using-parent
          ▼ ApiUrlResolver.obtainUserNamespace
            ‣ ApiUrlResolver.obtainUserNamespace - USER_NAMESPACE.length = 0, calling kubernetes API
            ‣ DriverHelper.getDriver
(node:526005) Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification.
(Use `node --trace-warnings ...` to show where the warning was created)
          ▼ ApiUrlResolver.obtainUserNamespace - kubeapi success: admin-devspaces
            ‣ DriverHelper.getDriver
          ▼ TestWorkspaceUtil.waitWorkspaceStatus
          ▼ ApiUrlResolver.obtainUserNamespace - admin-devspaces
            ‣ DriverHelper.getDriver
          ▼ TestWorkspaceUtil.stopWorkspaceByName - sample-using-parent stopped successfully
          ▼ TestWorkspaceUtil.deleteWorkspaceByName - sample-using-parent
          ▼ ApiUrlResolver.obtainUserNamespace - admin-devspaces
            ‣ DriverHelper.getDriver
            ‣ DriverHelper.getDriver
          ▼ TestWorkspaceUtil.deleteWorkspaceByName - sample-using-parent deleted successfully
          ▼     at /home/mmusiien/che-projects/che/tests/e2e/specs/MochaHooks.ts:39:12 - delete workspace name


  5 passing (2m)
```

### How to test this PR?
Run the WorkspaceWithParent.spec.ts as usual
 -->

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix or new feature ](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix-or-new-feature)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by SkorikSergey — APPROVED** [review]
Approved. Setting ` `security.workspace.trust.enabled: false`  is much better than constantly fixing trustDialog errors.

**Review by artaleks9 — APPROVED** [review]

File: tests/e2e/specs/miscellaneous/WorkspaceWithParent.spec.ts [hunk 1]
```diff
@@ -39,12 +38,11 @@ suite(`Workspace using a parent test suite ${BASE_TEST_CONSTANTS.TEST_ENVIRONMEN
 		CLASSES.KubernetesCommandLineToolsExecutor
 	);
 	const driverHelper: DriverHelper = e2eContainer.get(CLASSES.DriverHelper);
-	const createWorkspace: CreateWorkspace = e2eContainer.get(CLASSES.CreateWorkspace);
-
 	let podName: string = '';
 
 	suiteSetup(function (): void {
 		kubernetesCommandLineToolsExecutor.loginToOcp();
+		kubernetesCommandLineToolsExecutor.namespace = BASE_TEST_CONSTANTS.TEST_NAMESPACE;
```
Comment (dmytro-ndp) [inline tests/e2e/specs/miscellaneous/WorkspaceWithParent.spec.ts]: BASE_TEST_CONSTANTS.TEST_NAMESPACE could be empty, e.g. when run Dev Spaces test pipelines.

There is an example of code which handles absence of BASE_TEST_CONSTANTS.TEST_NAMESPACE: [KubernetesCommandLineToolsExecutor](https://github.com/eclipse-che/che/blob/main/tests/e2e/utils/KubernetesCommandLineToolsExecutor.ts#L49-L58)

Comment (musienko-maxim) [reply tests/e2e/specs/miscellaneous/WorkspaceWithParent.spec.ts]: I removed it. Thx for pointing! Seems, unnecessary operation after experimenting with  test launching

Comment (dmytro-ndp) [reply tests/e2e/specs/miscellaneous/WorkspaceWithParent.spec.ts]: Thanks

**Review by dmytro-ndp — APPROVED** [review]
LGTM

**Pull request merged**